fed/proxy/grpc_proxy.py

```diff
@@ -0,0 +1,311 @@
+import asyncio
+import copy
+import cloudpickle
+import grpc
+import logging
+import threading
+import json
+import importlib.metadata
+from typing import Dict
+
+
+import fed.utils as fed_utils
+
+from fed.config import CrossSiloCommConfig, CrossSiloGrpcCommConfig
+from fed._private.compatible_utils import _compare_version_strings
+from fed._private.grpc_options import _DEFAULT_GRPC_CHANNEL_OPTIONS, _GRPC_SERVICE
```

Comment:
I'm not sure if it's a good practice that we introduce the `grpc_options` in the `_private` package, because grpc_options is also the implementation details.

Reply:
We can move `grpc_options`here.